tools/tplgtool2.py

```diff
@@ -864,6 +864,34 @@ def node_sublabel(self, widget: Container) -> str:
             sublabel = ""
         return sublabel
 
+    PCM_NUM_REGEX = re.compile(r"PCM(?P<num>\d+)", re.IGNORECASE)
+    HOST_COPIER_NUM_REGEX = re.compile(r"host-copier\.(?P<num>\d+)", re.IGNORECASE)
+    def pcm_name(self, widget: Container):
+        """The widget argument is expected to be a PCMnnnX where nnn is
+        a pcm_id.  Find in self._tplg.pcm_list the corresponding PCM and
+        return its pcm_name.
+
+        """
+        wname = widget.name
+
+        match = self.HOST_COPIER_NUM_REGEX.match(wname)
+        if match is None:
+            match = self.PCM_NUM_REGEX.match(wname)
+
+        if match is None:
+            # Don't crash the entire program when passed a wrong input;
+            # put an error message in the graph instead.
+            return f'No PCM or host-copier number found in "{wname}"'
+
+        pcm_id = int(match["num"])
+        pcm_names = [ pcm.pcm_name for pcm in self._tplg.pcm_list if pcm.pcm_id == pcm_id ]
+
+        n_l = len(pcm_names)
+        # A duplicate pcm_id is a very severe issue
+        assert n_l <= 1, f"Unexpected number of PCMs matching pcm_id={pcm_id}: {n_l} ({pcm_names})"
+
+        return pcm_names[0] if n_l == 1 else f"No PCM with ID={pcm_id}"
+
```

Comment:
IIRC @ranj063 already did some changes in the topology2 naming, not sure if these changes are based on top of the topology2 changes?

In addition, while the name of the PCM device is nice, the PCM device number is the most useful for testing. How difficult would it be to add the hw: ID?

Reply:
> IIRC @ranj063 already did some changes in the topology2 naming, not sure if these changes are based on top of the topology2 changes?
> 
> In addition, while the name of the PCM device is nice, the PCM device number is the most useful for testing. How difficult would it be to add the hw: ID?

@plbossart I think this is a good change on top of what I did with the naming. With a name like host-copier.1.playback you've already got the PCM ID i.e. 1 and this adds the name of the PCM next to it